appkit/src/test/scala/org/ergoplatform/appkit/DHTProverSpec.scala

```diff
@@ -0,0 +1,53 @@
+package org.ergoplatform.appkit
+
+import org.ergoplatform.{ErgoAddressEncoder, Pay2SAddress}
+import org.scalatest.{Matchers, PropSpec}
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+import scorex.util.encode.Base16
+import sigmastate.eval._
+import sigmastate.interpreter.CryptoConstants
+import sigmastate.serialization.ErgoTreeSerializer
+import special.sigma.GroupElement
+
+class DHTProverSpec extends PropSpec with Matchers
+  with ScalaCheckDrivenPropertyChecks
+  with AppkitTesting
+  with HttpClientTesting {
+
+  property("DHTProver") {
+    val ergoClient = createMockedErgoClient(MockData(Nil, Nil))
+    val g: GroupElement = CryptoConstants.dlogGroup.generator
+    val x = BigInt("187235612876647164378132684712638457631278").bigInteger
+    val y = BigInt("340956873409567839086738967389673896738906").bigInteger
+    val gX:GroupElement = g.exp(x)
+    val gY:GroupElement = g.exp(y)
+    val gXY:GroupElement = gY.exp(x)
+
+    ergoClient.execute { ctx: BlockchainContext =>
+      val input = ctx.newTxBuilder.outBoxBuilder.registers(
+        ErgoValue.of(gY), ErgoValue.of(gXY) // <--- correct one, (Alice's full mix box)
+        //        ErgoValue.of(gXY), ErgoValue.of(gY) // <--- wrong one, (Bob's full mix box). Alice should not be able to sign.
+      ).value(20000000).contract(ctx.compileContract(
+        ConstantsBuilder.empty(),
+        """{
+          |  val gY = SELF.R4[GroupElement].get
+          |  val gXY = SELF.R5[GroupElement].get
+          |  proveDHTuple(gY, gY, gXY, gXY)
+          |}""".stripMargin
+      )).build().convertToInputWith("f9e5ce5aa0d95f5d54a7bc89c46730d9662397067250aa18a0039631c0f5b809", 0)
+
+      val txB = ctx.newTxBuilder()
+      val output = txB.outBoxBuilder().value(10000000).contract(ctx.compileContract(ConstantsBuilder.empty(),"{sigmaProp(true)}")).build()
+      val inputs = new java.util.ArrayList[InputBox]()
+      inputs.add(input)
+
+      val tree = "100207036ba5cfbc03ea2471fdf02737f64dbcd58c34461a7ec1e586dcd713dacbf89a120400d805d601db6a01ddd6027300d603b2a5730100d604e4c672030407d605e4c672030507eb02ce7201720272047205ce7201720472027205"
```

Comment:
Please add a comment with the original code of the tree.

Reply:
This was the ergo tree of some randomly selected box from explorer. I've added the box Id in a comment.